### PR TITLE
Chore/add wf integration tests

### PR DESCRIPTION
**Pull request by haraprasadj**

### CODE REVIEW

File: .github/workflows/integration_tests.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                poetry run python -m gen3_ci.scripts.select_ci_environment
+              fi
+
+          # TODO: Improve the logic to do differential updates to the env, not roll all services
+          # Apply the changes to the manifest of the selected CI environment, roll the pods and run usersync
+          # Generate API keys for test users for the environment
+          - name: Prepare CI environment
+            run: |
+              mkdir $HOME/.gen3
+              poetry run python -m gen3_ci.scripts.prepare_ci_environment
```
Comment (Avantol13) [inline .github/workflows/integration_tests.yaml:83]: some general feedback on the underlying logs here:

- There's a lot of `Waiting for job completion...` it'd be nice if the log had the jenkins job that it was waiting on too for debugging purposes

File: .github/workflows/integration_tests.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+          - name: Run tests
+            run: poetry run pytest --html=output/report.html --self-contained-html -n auto -m "not wip"
+
+          - name: Archive test report
+            uses: actions/upload-artifact@v3
+            with:
+              name: test-report
+              path: ~/**/output/report.html
```
Comment (Avantol13) [inline .github/workflows/integration_tests.yaml:92]: at the moment this isn't a super useful report (just tests passed/failed)

One thing to consider is how we mimic some of features of current Jenkins integration testing (like screenshots for UI tests and getting a dump of the service logs for when things go wrong).

I don't think you have to solve that right now, but we will want a way to easily get to service logs when the integration tests fail

**Review by Avantol13 — CHANGES_REQUESTED** [review]
This is amazing :tada: super excited about all this!

Some minor things regarding the underlying vigil testing repo (which isn't really the focus of this review, but I figured I'd put some notes here).

I'd really love to follow the same SDLC and dev practices in the vigil repo as the rest of the code. Eventually enforcing required reviewers on all PRs, etc.

In terms of code organization, things are looking great, I like that the Jenkins tests are in the repo as well. I also appreciate the use of logging, though I would never be mad about more logs in places where it might be helpful.

The only thing I'd recommend (that I'm a stickler for in PR reviews for services) is the use of docstrings in all non-private functions/classes to better document the expected args, types, returns, and general purpose of the function. We use the Google style for docstrings: https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings

File: .github/workflows/integration_tests.yaml (hunk 1 above)
Comment (haraprasadj) [reply .github/workflows/integration_tests.yaml:83]: Agree, already made a note of this, will add more logs on the job details

File: .github/workflows/integration_tests.yaml (hunk 2 above)
Comment (haraprasadj) [reply .github/workflows/integration_tests.yaml:92]: Not sure if you noticed the `show details` next to the result ... there are step logs as well in there once expanded. But yes, still looking for a better reporting mechanism.

**Review by Avantol13 — APPROVED** [review]

**Pull request merged**